python/setup.py

```diff
@@ -132,6 +135,12 @@ def initialize_options(self):
             if not hasattr(sys, 'gettotalrefcount'):
                 self.build_type = 'release'
 
+        self.with_arrow_gpu = strtobool(
+            os.environ.get('PYARROW_WITH_ARROW_GPU', '0'))
```

Comment:
`PYARROW_WITH_GPU` sounds simpler (compare with `PYARROW_WITH_PLASMA` or `PYARROW_WITH_ORC`).

Reply:
Sounds good.

Reply:
Renamed `PYARROW_WITH_ARROW_GPU` to `PYARROW_WITH_CUDA`